pkg/operator/encryption/shared_encryption.go

```diff
@@ -581,6 +630,18 @@ func getGRsActualKeys(encryptionConfig *apiserverconfigv1.EncryptionConfiguratio
 			}
 		}
 
+		// now we can explicitly handle the fake identity provider based on the key state
+		switch {
+		case grk.writeKey.mode == identity && hasFakeIdentityProvider:
+			grk.writeKey.key = lastProvider.AESGCM.Keys[0]
```

Comment:
what will do the migration controller when it sees a write key like that?

Reply:
Treat it just like an other write key and run a storage migration if needed.